### PR TITLE
[MARVP-44] Offine data functionality

### PR DESCRIPTION
**Pull request by krishk95**
Related to : https://konysolutions.atlassian.net/browse/MARVP-44

### CODE REVIEW

File: actions/mobile/AS_AppEvents_h334c68ef261418898ed0881d70a84cb.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        "type": "ACTION_SEQUENCE",
+        "params": [{
+            "datatype": "collection",
+            "name": "eventobject"
+        }],
+        "language": "js",
+        "actions": [{
+            "id": "ADD_SNIPPET__ccfe68ae5bea453eae5a240dd0b9f459",
+            "type": "ADD_SNIPPET",
+            "codeSnippet": "var appEventData = accelerateEventData.eventdata;\nvar updatedData = {\n  \"data\":\"Sample Test\"\n} \naccelerateEventData.eventdata = updatedData;\nalert(accelerateEventData.eventdata);",
```
Comment (mishradileep) [inline actions/mobile/AS_AppEvents_h334c68ef261418898ed0881d70a84cb.json]: No ```"Sample Data"``` please

File: actions/mobile/appEvents.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+define({
+    AS_AppEvents_hf9bdba361244e8884a25e0b647863b6: function AS_AppEvents_hf9bdba361244e8884a25e0b647863b6(eventobject) {
+        var self = this;
```
Comment (mishradileep) [inline actions/mobile/appEvents.js:3]: self not used. Please remove

File: binaries/buildLogs/logfile.log [hunk 3]
```diff
@@ -0,0 +1,3694 @@
+  Started by upstream project "CloudBuildService/Visualizer/Builds/cloudBuildVisualizerApp" build number 1298
```
Comment (mishradileep) [inline binaries/buildLogs/logfile.log]: Add log file in gitignore

File: modules/eventSessionData.js [hunk 4]
```diff
@@ -0,0 +1,238 @@
+accelerateSessionData = {
+  "eventSessionData":{
+    "records":[
+    {
+        "event_id":3,"event_session_id":2,"session_end_date":"2019-09-04T08:45:00Z",
+        "session_name":"Breakfast and Registration",
+        "session_start_date":"2019-09-04T07:30:00Z",
+        "session_code":"9OOY1C",
+        "CreatedDateTime":"2019-05-16T09:47:45Z",
+        "LastUpdatedDateTime":"2019-05-16T09:47:45Z"
```
Comment (mishradileep) [inline modules/eventSessionData.js:10]: Lots of details of session is missing? Can't we make NULL value with all possible keys

File: modules/eventSessionData.js [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+    }
+,
+    {
+        "event_id":3,"event_session_id":3,"session_desc":"The evolution of digital technology today is never-ending. ",
+        "session_end_date":"2019-09-04T09:30:00Z",
+        "session_name":"Welcome and Kony Vision",
+        "session_start_date":"2019-09-04T09:00:00Z",
+        "session_long_desc":"Our mission is helping our customers keep up with the frenetic pace of digital and how our experience and expertise in the digital space helps companies excel at digital today and into the future.\n",
+        "session_code":"52Y1MX",
```
Comment (mishradileep) [inline modules/eventSessionData.js:19]: Seems incorrect data present.

File: modules/eventSessionData.js [hunk 6]
```diff
@@ -0,0 +1,238 @@
+accelerateSessionData = {
+  "eventSessionData":{
+    "records":[
+    {
+        "event_id":3,"event_session_id":2,"session_end_date":"2019-09-04T08:45:00Z",
```
Comment (mishradileep) [inline modules/eventSessionData.js:5]: Format properly

File: modules/eventStaticData.js [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+            "LastUpdatedBy":"",
+            "CreatedDateTime":"2019-07-26T07:51:45.518Z",
+            "LastUpdatedDateTime":"2019-07-30T10:48:19Z",
+            "SoftDeleteFlag":false
+        }
+]
+    }
+],"opstatus":0,"httpStatusCode":0
+}
+};
```
Comment (mishradileep) [inline modules/eventStaticData.js:56]: Need formatting

File: modules/fetchEventData.js [hunk 8]
```diff
@@ -0,0 +1,108 @@
+	
+	
+
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Unnecessary blank lines.
Missing function comment

File: modules/fetchEventData.js [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+    function syncEventData() {
+      	let isTimeStamp = kony.store.getItem("isTimeStampUpdated");
+      	if(isTimeStamp === null || isTimeStamp === undefined)
+      		kony.store.setItem("clientLastUpdatedTime","2019-08-04T15:27:26Z");
+        fetchObjectData(eventConstants.OBJECT_SERVICE_NAME, eventConstants.DATA_SYNC_OBJECT, {}, dataSyncFetchSuccess, dataSyncFetchFailure);
+    }
+ 
+    function dataSyncFetchSuccess(successResponse) {
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Missing function comment

File: modules/fetchEventData.js [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+            let serverLastUpdatedTime = records[0].LastUpdatedDateTime;
+          	let clientLastUpdatedTime = kony.store.getItem("clientLastUpdatedTime");
+            if (serverLastUpdatedTime > clientLastUpdatedTime) {
+                latestTimeStamp = serverLastUpdatedTime;
+                fetchEventStaticData();
+            }
+        }
+    }
+	
+	function dataSyncFetchFailure(failureResponse){
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Missing function comment

File: modules/fetchEventData.js [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+       	kony.print("Error occured in fetching the event data");
+        kony.print("Error occured is" + JSON.stringify(failureResponse));
+    }
+
+
+    function fetchEventStaticData() {
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Missing function comment

File: modules/fetchEventData.js [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+	function parseJSONResponse(successResponse){
+      let recordsLength = successResponse.hasOwnProperty("records") ? successResponse.records.length : -1;
+      if(recordsLength > 0){
+        for(let index = 0 ; index < recordsLength ; index++){
+          let currentRecord = successResponse.records[index];
+          currentRecord.sessionsList = JSON.parse(currentRecord.sessionsList);
+          currentRecord.tracks = JSON.parse(currentRecord.tracks);
+        }
+      }
+    }
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Missed function comment for all the function for this file

File: modules/fetchEventData.js [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+            kony.store.setItem("clientLastUpdatedTime",latestTimeStamp);
+          	kony.store.setItem("isTimeStampUpdated",true);
+        }
+    }
+
+    function speakerDataFetchFailure(failureResponse) {
+        kony.print("Error occured in fetching the event data");
+        kony.print("Error occured is" + JSON.stringify(failureResponse));
+    }
+
```
Comment (mishradileep) [inline modules/fetchEventData.js:190]: Remove blank like

File: modules/fetchEventData.js [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+    function syncEventData() {
+      	let isTimeStamp = kony.store.getItem("isTimeStampUpdated");
+      	if(isTimeStamp === null || isTimeStamp === undefined)
+      		kony.store.setItem("clientLastUpdatedTime","2019-08-04T15:27:26Z");
+        fetchObjectData(eventConstants.OBJECT_SERVICE_NAME, eventConstants.DATA_SYNC_OBJECT, {}, dataSyncFetchSuccess, dataSyncFetchFailure);
+    }
+ 
+    function dataSyncFetchSuccess(successResponse) {
+        let records = (successResponse.hasOwnProperty("records")) ? successResponse.records : null;
+        if (records !== null) {
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Best practice is to keep constant in left in compare operators.
```if (null !=== records) {```

This is applicable for whole

File: modules/fetchEventData.js [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+            if (serverLastUpdatedTime > clientLastUpdatedTime) {
+                latestTimeStamp = serverLastUpdatedTime;
+                fetchEventStaticData();
+            }
+        }
+    }
+	
+	function dataSyncFetchFailure(failureResponse){
+       	kony.print("Error occured in fetching the event data");
+        kony.print("Error occured is" + JSON.stringify(failureResponse));
```
Comment (mishradileep) [inline modules/fetchEventData.js]:
```
kony.print("Something went wrong while fetching the event data");
kony.print("Error details: " + JSON.stringify(failureResponse));
```

File: modules/fetchEventData.js [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+
+    function fetchEventStaticData() {
+        var queryParams = {
+            "$filter": "((SoftDeleteFlag ne true) or (SoftDeleteFlag eq null))",
+            "$expand": "event_inner_location,wifi_info,location"
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Always put space after comma (;)

File: modules/fetchEventData.js [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+        if (records !== null) {
+            accelerateEventData.eventdata = successResponse;
+            kony.print(accelerateEventData.eventdata);
+            fetchEventSessionData();
+        }
+    }
+
+    function eventDataFetchFailure(failureResponse) {
+        kony.print("Error occured in fetching the event data");
+        kony.print("Error occured is" + JSON.stringify(failureResponse));
```
Comment (mishradileep) [inline modules/fetchEventData.js]:
```
kony.print("Something went wrong while fetching the event data");
kony.print("Error details: " + JSON.stringify(failureResponse));

File: modules/fetchEventData.js [hunk 18]
```diff
@@ -0,0 +1,108 @@
...
+        if (records !== null) {
+            accelerateSessionData.eventSessionData = successResponse;
+            kony.print(accelerateSessionData.eventSessionData);
+            fetchSpeakersData();
+        }
+    }
+
+    function eventSessionDataFetchFailure(failureResponse) {
+        kony.print("Error occured in fetching the event data");
+        kony.print("Error occured is" + JSON.stringify(failureResponse));
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Same as on #40

File: modules/fetchEventData.js (hunk 12 above)
Comment (mishradileep) [reply modules/fetchEventData.js]: EOF is missing

File: modules/fetchEventData.js [hunk 19]
```diff
@@ -0,0 +1,108 @@
...
+            kony.print(accelerateSpeakerData.eventSpeakerData);
+          	parseJSONResponse(successResponse);
+            kony.store.setItem("clientLastUpdatedTime",latestTimeStamp);
+          	kony.store.setItem("isTimeStampUpdated",true);
+        }
+    }
+
+    function speakerDataFetchFailure(failureResponse) {
+        kony.print("Error occured in fetching the event data");
+        kony.print("Error occured is" + JSON.stringify(failureResponse));
```
Comment (mishradileep) [inline modules/fetchEventData.js]: Same as on #40

File: modules/fetchEventData.js (hunk 12 above)
Comment (mishradileep) [reply modules/fetchEventData.js]: You need indentation on indentation

**Review by mishradileep — CHANGES_REQUESTED** [review]
Look into inline comments

Global comments
Please look into indentation
Make sure function comments are present for all the function

File: actions/mobile/appEvents.js (hunk 2 above)
Comment (krishk95) [reply actions/mobile/appEvents.js:3]: This is the generated code.

**Review by vijaysamanthapudi — APPROVED** [review]

**Review by vigneshwarankarunanithi — APPROVED** [review]

**Review by mishradileep — APPROVED** [review]

**Pull request merged**